core/src/main/java/org/elasticsearch/index/query/CommonTermsQueryBuilder.java

```diff
@@ -280,78 +277,79 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
                         }
-                    } else {
-                        throw new ParsingException(parser.getTokenLocation(), "[" + CommonTermsQueryBuilder.NAME +
-                                "] query does not support [" + currentFieldName + "]");
-                    }
-                } else if (token.isValue()) {
-                    if (parseContext.getParseFieldMatcher().match(currentFieldName, QUERY_FIELD)) {
-                        text = parser.objectText();
-                    } else if (parseContext.getParseFieldMatcher().match(currentFieldName, ANALYZER_FIELD)) {
-                        analyzer = parser.text();
-                    } else if (parseContext.getParseFieldMatcher().match(currentFieldName, DISABLE_COORD_FIELD)) {
-                        disableCoord = parser.booleanValue();
-                    } else if (parseContext.getParseFieldMatcher().match(currentFieldName, AbstractQueryBuilder.BOOST_FIELD)) {
-                        boost = parser.floatValue();
-                    } else if (parseContext.getParseFieldMatcher().match(currentFieldName, HIGH_FREQ_OPERATOR_FIELD)) {
-                        highFreqOperator = Operator.fromString(parser.text());
-                    } else if (parseContext.getParseFieldMatcher().match(currentFieldName, LOW_FREQ_OPERATOR_FIELD)) {
-                        lowFreqOperator = Operator.fromString(parser.text());
-                    } else if (parseContext.getParseFieldMatcher().match(currentFieldName, MINIMUM_SHOULD_MATCH_FIELD)) {
-                        lowFreqMinimumShouldMatch = parser.text();
-                    } else if (parseContext.getParseFieldMatcher().match(currentFieldName, CUTOFF_FREQUENCY_FIELD)) {
-                        cutoffFrequency = parser.floatValue();
-                    } else if (parseContext.getParseFieldMatcher().match(currentFieldName, AbstractQueryBuilder.NAME_FIELD)) {
-                        queryName = parser.text();
-                    } else {
-                        throw new ParsingException(parser.getTokenLocation(), "[" + CommonTermsQueryBuilder.NAME +
-                                "] query does not support [" + currentFieldName + "]");
                     }
                 }
-            }
-            parser.nextToken();
-        } else {
-            text = parser.objectText();
-            // move to the next token
-            token = parser.nextToken();
-            if (token != XContentParser.Token.END_OBJECT) {
-                throw new ParsingException(parser.getTokenLocation(),
-                        "[common] query parsed in simplified form, with direct field name, but included more options than just " +
-                                "the field name, possibly use its 'options' form, with 'query' element?");
+            } else {
+                fieldName = parser.currentName();
+                text = parser.objectText();
             }
         }
 
-        if (text == null) {
```

Comment:
I removed this of checks at the end of parsing as they are not needed. The query builder throws error anyways as we validate its input and we also test that in our tests.